documentation/src/main/asciidoc/user_guide/server_protocols_hotrod_transaction.adoc

```diff
@@ -0,0 +1,176 @@
+
+The {trc} is a {rc} extension with single method:
+
+* `getTransactionManager()`:
+it returns the {tm} in use by this cache.
+
+===== Usage Example
+
+**Using the configured `TransactionManager` and `TransactionMode`**
+
+[source,java]
+----
+//configure the transaction manager and transaction mode
+org.infinispan.client.hotrod.configuration.ConfigurationBuilder cb = new org.infinispan.client.hotrod.configuration.ConfigurationBuilder();
+cb.transaction().transactionManagerLookup(RemoteTransactionManagerLookup.getInstance());
+cb.transaction().transactionMode(TransactionMode.NON_XA);
+
+RemoteCacheManager rcm = new RemoteCacheManager(cb.build());
+
+//my-cache will be using the transaction manager and mode as configured above.
+TransactionalRemoteCache<String, String> cache = rcm.getTransactionalCache("my-cache");
+
+//it returns the transaction manager used by the cache
+TransactionManager tm = cache.getTransactionManager();
+
+//just a simple transaction.
+tm.begin();
+cache.put("k1", "v1");
+System.out.println("K1 value is " + cache.get("k1"));
+tm.commit();
+----
+
+**Overriding the `TransactionManager`**
+
+[source,java]
+----
+//configure the transaction manager and transaction mode
+org.infinispan.client.hotrod.configuration.ConfigurationBuilder cb = new org.infinispan.client.hotrod.configuration.ConfigurationBuilder();
+cb.transaction().transactionManagerLookup(RemoteTransactionManagerLookup.getInstance());
+cb.transaction().transactionMode(TransactionMode.NON_XA);
+
+RemoteCacheManager rcm = new RemoteCacheManager(cb.build());
+
+//a custom transaction manager
+TransactionManager myCustomTM = ...
+
+//my-cache overrides the TransactionManager. null for TransactionMode means to use the default.
+TransactionalRemoteCache<String, String> cache = rcm.getTransactionalCache("my-cache", myCustomTM, null);
+
+//just a simple transaction.
+myCustomTM.begin();
+cache.put("k1", "v1");
+System.out.println("K1 value is " + cache.get("k1"));
+myCustomTM.commit();
+----
+
+**Overriding the TransactionMode**
+
+[source,java]
+----
+//configure the transaction manager and transaction mode
+org.infinispan.client.hotrod.configuration.ConfigurationBuilder cb = new org.infinispan.client.hotrod.configuration.ConfigurationBuilder();
+cb.transaction().transactionManagerLookup(RemoteTransactionManagerLookup.getInstance());
+cb.transaction().transactionMode(TransactionMode.NON_XA);
+
+RemoteCacheManager rcm = new RemoteCacheManager(cb.build());
+
+//my-cache overrides the TransactionMode.
+TransactionalRemoteCache<String, String> cache = rcm.getTransactionalCache("my-cache", null, TransactionMode.NON_DURABLE_XA);
+
+//it returns the transaction manager used by the cache
+TransactionManager tm = cache.getTransactionManager();
+
+//just a simple transaction.
+tm.begin();
+cache.put("k1", "v1");
+System.out.println("K1 value is " + cache.get("k1"));
+tm.commit();
+----
+
+//recovery not yet implemented.
```

Comment:
Remove this line and amend the paragraph above where you list the possible tx modes to indicate which ones work and which ones don't yet.